source/UIABrowseMode.py

```diff
@@ -152,27 +152,34 @@ def isChild(self,parent):
 		return self.level>parent.level
 
 def UIAHeadingQuicknavIterator(itemType,document,position,direction="next"):
-	if position:
-		curPosition=position
+	reverse = bool(direction == "previous")
+	entireDocument = document.makeTextInfo(textInfos.POSITION_ALL)
+	if position is None:

```

Comment:
@leonardder  position will be None if called by the Elements list to list all headings. Position is a TextInfo if called by quicknav to jump from a specific location.

Reply:
Thanks for clarifying!